physical/postgresql/postgresql.go

```diff
@@ -238,6 +257,14 @@ func NewPostgreSQLBackend(conf map[string]string, logger log.Logger) (physical.B
 		upsert_function: quoted_upsert_function,
 	}
 
+	// Determine if we should participate in leadership elections. This can
+	// be changed at runtime but is not persisted.
+	nonVoter, err := permanentStandby(conf)
```

Comment:
I'm not quite sure how to test this feature in CI. I think we'll want a replicated PGSQL environment, which I can set up with docker-compose, and then we'll also need the ability to modify a node's config and restart it. The restart part is difficult; the dockerized testcluster would let us do full binary testing (I'm not sure if it supports config file snippets but we could theoretically add it and PostgreSQL support), but we can't reliably restart these nodes with different configs after they were initially started up...